src/Extension/TextNode.php

```diff
@@ -11,14 +11,15 @@ protected function getGrammarGrammarSequence()
 
     protected function _buildSequenceItem(&$grammar, $sequenceItem, $grammarParser, $options)
     {
-        if (!empty($options['caseInsensitive'])) {
+        if ($options['caseInsensitive']) {
             $regex = \ParserGenerator\RegexUtil::buildRegexFromString((string)$sequenceItem->getSubnode(0)->getValue());
-            return new \ParserGenerator\GrammarNode\Regex($regex, !empty($options['ignoreWhitespaces']),
-                !empty($options['caseInsensitive']));
-        } elseif (empty($options['ignoreWhitespaces'])) {
```

Comment:
Took the opportunity and unfangled the `elseifs` a bit, making the structure clearer to follow